src/extension.ts

```diff
@@ -36,25 +36,25 @@ import {
 import { LazyProgress } from "./lazy-progress";
 import * as fs from "fs";
 import * as semver from "semver";
+import { getJavaHome } from "./getJavaHome";
+
+const outputChannel = window.createOutputChannel("Metals");
 
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
 
-  const userJavaHome = workspace.getConfiguration("metals").get("javaHome");
-  if (typeof userJavaHome === "string" && userJavaHome !== "") {
-    fetchAndLaunchMetals(context, userJavaHome);
-  } else {
-    require("find-java-home")((err, javaHome) => {
-      if (err) window.showErrorMessage("Unable to find Java home.");
-      fetchAndLaunchMetals(context, javaHome);
+  getJavaHome()
+    .then(javaHome => fetchAndLaunchMetals(context, javaHome))
+    .catch(err => {
+      const message = "Unable to detect Java home";
```

Comment:
Update message?

> Unable to find Java 8 home. To fix this problem, update the 'Java Home' setting in Metals to point to a Java 8 home directory.

Reply:
![image](https://user-images.githubusercontent.com/691940/49878352-aaf28400-fe27-11e8-9e18-fa00db84cabf.png)

Here it is

Reply:
😍